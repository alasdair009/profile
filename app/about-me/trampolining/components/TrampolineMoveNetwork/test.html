<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
