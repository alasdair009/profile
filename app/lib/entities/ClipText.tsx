"use client";

import styled from "styled-components";
import {rem} from "polished";

export const ClipText = styled.span`
  align-items: center;
  //background: black;
  display: flex;
  font-size: ${rem(56)};
  height: 100%;
  justify-content: center;
  left: 50%;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  -webkit-text-fill-color: transparent;
  text-fill-color: transparent;
  -webkit-background-clip: text;
  background-clip: text;
  background: black;
  color: white;
`;