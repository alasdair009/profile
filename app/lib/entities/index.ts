"use client";

import styled from "styled-components";

export * from "@/entities";
export { Splash } from "./Splash";

export const FireWrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  position: absolute;
  width: 100%;
`;
export const FirePadding = styled.div<{ $backgroundColor: string }>`
  background: ${({ $backgroundColor }) => $backgroundColor};
  flex: 1;
`;
