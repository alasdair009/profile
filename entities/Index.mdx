import { Meta } from "@storybook/blocks";
import Image from "next/image";
import amLogo from "./assets/am.svg";
import { rem } from "polished";

<Meta title="Alasdair Macrae Design" />

# Alasdair Macrae Design

<Image
  src={amLogo}
  alt="Green and white AM letters"
  width={0}
  height={0}
  style={{ maxWidth: rem(240), width: "100%", height: "auto" }}
/>

I am a keen supported of Storybook js using it across the majority of projects I build from websites, web-apps and even desktop applications.
