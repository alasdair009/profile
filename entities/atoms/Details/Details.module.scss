@use "../../../app/styles/colors";
@use "../../../app/styles/dimensions";
@use "../../../app/styles/typography";
@use "../../../app/styles/helpers/rem";
@use "sass:map";

.root {
  margin: 0 auto rem.toRem(1);
  max-width: rem.toRem(dimensions.$globalContentMaxWidth);
  overflow: hidden;
  position: relative;
  width: 100%;
}

.summary {
  align-items: center;
  background: colors.$colorGreyLight;
  cursor: pointer;
  font-size: typography.getFontSize(mlarge);
  justify-content: flex-start;
  padding: dimensions.getSize(s8);
  position: relative;
  width: 100%;

  &:hover {
    filter: brightness(1.3);
  }

  &::marker {
    color: colors.$colorGreenGrass;
  }
}

.content {
  background: colors.$colorGreyDark;
  overflow: hidden;
  padding: dimensions.getSize(s8);
  position: relative;
  width: 100%;
}
