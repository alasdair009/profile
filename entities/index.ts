export { borderRadii } from "./design-tokens/effects";
export { colors } from "./design-tokens/colors";
export { fonts } from "./design-tokens/typography";
export { sizes } from "./design-tokens/dimensions";

export { Button } from "./atoms/Button";
export { Fire } from "./atoms/Fire";
export { Heading } from "./atoms/Heading";
export { Link } from "./atoms/Link";
export { Orb } from "./atoms/Orb";
export { Paragraph } from "./atoms/Paragraph";
export { Spacer } from "./atoms/Spacer";
export { TextMask } from "./atoms/TextMask";

export { Container } from "./organisms/Container";
export { ContentPlate } from "./organisms/ContentPlate";
export { Header } from "./organisms/Header";
export { Splash } from "./organisms/Splash";
