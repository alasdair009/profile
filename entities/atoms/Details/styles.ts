"use client";
import styled from "styled-components";
import { colors, fontSizes, globalContentMaxWidth, sizes } from "@/entities";
import { rem } from "polished";

export const Root = styled.details`
  margin: 0 auto ${rem(1)};
  max-width: ${rem(globalContentMaxWidth)};
  overflow: hidden;
  position: relative;
  width: 100%;
`;

export const Summary = styled.summary`
  align-items: center;
  background: ${colors.greyLight};
  cursor: pointer;
  font-size: ${fontSizes.mlarge.rem};
  justify-content: flex-start;
  padding: ${sizes.s8.rem};
  position: relative;
  width: 100%;

  &:hover {
    filter: brightness(1.3);
  }

  &::marker {
    color: ${colors.greenGrass};
  }
`;

export const Content = styled.div`
  background: ${colors.greyDark};
  overflow: hidden;
  padding: ${sizes.s8.rem};
  position: relative;
  width: 100%;
`;
