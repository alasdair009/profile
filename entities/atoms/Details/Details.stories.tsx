import { Details } from "./Details";
import { Meta, StoryObj } from "@storybook/nextjs";
import { expect, waitFor, within } from "storybook/test";
import { Heading, Paragraph } from "@/entities";

const meta: Meta<typeof Details> = {
  component: Details,
  argTypes: {
    summary: {
      type: "string",
    },
  },
  args: {
    summary: "Title for the details element",
    children: (
      <>
        <Heading>Content</Heading>
        <Paragraph>
          Some copy to display with the Heading and whatever else takes your
          fancy.
        </Paragraph>
      </>
    ),
  },
};
export default meta;

export const Default: StoryObj<typeof Details> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);

    await waitFor(() =>
      expect(canvas.getByText(args.summary)).toBeInTheDocument()
    );
  },
};

export const Accordion: StoryObj<typeof Details> = {
  args: {
    name: "accordion",
  },
  render: (args) => (
    <>
      <Details summary="First details" name={args.name}>
        {args.children}
      </Details>
      <Details summary="Second details" name={args.name}>
        {args.children}
      </Details>
      <Details summary="Third details" name={args.name}>
        {args.children}
      </Details>
    </>
  ),
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);

    await waitFor(() =>
      expect(canvas.getByText("First details")).toBeInTheDocument()
    );
    await waitFor(() =>
      expect(canvas.getByText("Second details")).toBeInTheDocument()
    );
    await waitFor(() =>
      expect(canvas.getByText("Third details")).toBeInTheDocument()
    );
  },
};
