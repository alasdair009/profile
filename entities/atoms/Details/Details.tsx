import { CSSProperties, HTMLAttributes } from "react";
import styles from "./Details.module.scss";

type DetailsProps = {
  /**
   * Content of the summary
   */
  summary: string;
  /**
   * Name of the element. Use the same name to create Accordions where only one is open.
   */
  name?: string;
  /**
   * Duration of the open/close transition.
   */
  transitionDuration?: number;
} & HTMLAttributes<HTMLDetailsElement>;

/**
 * Text to display when there is a problem the user needs to address or be aware of.
 */
export function Details({
  summary,
  transitionDuration = 0.5,
  style,
  children,
  ...rest
}: DetailsProps) {
  return (
    <details
      className={styles.root}
      data-testid={Details.displayName}
      style={
        {
          ...style,
          "--transition-duration": transitionDuration,
        } as CSSProperties
      }
      {...rest}
    >
      <summary className={styles.summary}>{summary}</summary>
      <div className={styles.content}>{children}</div>
    </details>
  );
}
Details.displayName = "Details";
