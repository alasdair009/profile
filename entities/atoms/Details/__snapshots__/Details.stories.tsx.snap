// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Details Accordion play-test 1`] = `
<details data-testid="Details"
         name="accordion"
         class="sc-ezTrPE gcihre"
>
  <summary class="sc-drnuxz lcxFEB">
    First details
  </summary>
  <div class="sc-hDcvty dBapbO">
    <h1 data-testid="Heading"
        class="sc-eJgwjL eUOrbm"
    >
      Content
    </h1>
    <p data-testid="Paragraph"
       class="sc-gutikT beCvGK"
    >
      Some copy to display with the Heading and whatever else takes your fancy.
    </p>
  </div>
</details>
<details data-testid="Details"
         name="accordion"
         class="sc-ezTrPE gcihre"
>
  <summary class="sc-drnuxz lcxFEB">
    Second details
  </summary>
  <div class="sc-hDcvty dBapbO">
    <h1 data-testid="Heading"
        class="sc-eJgwjL eUOrbm"
    >
      Content
    </h1>
    <p data-testid="Paragraph"
       class="sc-gutikT beCvGK"
    >
      Some copy to display with the Heading and whatever else takes your fancy.
    </p>
  </div>
</details>
<details data-testid="Details"
         name="accordion"
         class="sc-ezTrPE gcihre"
>
  <summary class="sc-drnuxz lcxFEB">
    Third details
  </summary>
  <div class="sc-hDcvty dBapbO">
    <h1 data-testid="Heading"
        class="sc-eJgwjL eUOrbm"
    >
      Content
    </h1>
    <p data-testid="Paragraph"
       class="sc-gutikT beCvGK"
    >
      Some copy to display with the Heading and whatever else takes your fancy.
    </p>
  </div>
</details>
`;

exports[`atoms/Details Default play-test 1`] = `
<details data-testid="Details"
         class="sc-ezTrPE gcihre"
>
  <summary class="sc-drnuxz lcxFEB">
    Title for the details element
  </summary>
  <div class="sc-hDcvty dBapbO">
    <h1 data-testid="Heading"
        class="sc-eJgwjL eUOrbm"
    >
      Content
    </h1>
    <p data-testid="Paragraph"
       class="sc-gutikT beCvGK"
    >
      Some copy to display with the Heading and whatever else takes your fancy.
    </p>
  </div>
</details>
`;
